src/components/OrgDirectoryGrid.tsx

```diff
@@ -1,8 +1,15 @@
 import DirectoryOrgs from './DirectoryOrgs';
 import { api } from '@src/trpc/server';
+import { getServerAuthSession } from '@src/server/auth';
 
 const OrgDirectoryGrid = async ({ tag }: { tag?: string }) => {
-  const clubs = await api.club.all.query({ tag });
+  const session = await getServerAuthSession();
+  let clubs;
+  if(session){
+    clubs = await api.club.allButJoined.query({ tag });
+  } else{
+    clubs = await api.club.all.query({ tag });
+  }
```

Comment:
Try moving this logic into the server. COmbine the `all` and `allButJoined` into one query